apps/extension/src/core/media/openrouter.ts

```diff
@@ -29,16 +29,17 @@ export function init(
           user: meta.user_identifier ?? undefined,
           num_outputs: num_generations,
           num_inference_steps: num_inference_steps,
-          mime_type: mime_type ?? undefined,
+          extension: extension ?? undefined,
         }
       },
       transformResponse: (res) => {
         const anyRes = res as any
-        return anyRes["data"]["uris"].map((g: string) => {
+        return anyRes["data"].map(({uri, url}: {uri: string, url: string | null}) => {
           return {
-            uri: g
+            uri: uri,
+            url: url || null // url is null if the output is not hosted
```

Comment:
no need for "|| null" if we already have that assumed by the type above. you'd just do `return {uri, url}`

Reply:
changed